Lib/asyncio/tasks.py

```diff
@@ -627,7 +627,8 @@ def ensure_future(coro_or_future, *, loop=None):
         return task
     elif futures.isfuture(coro_or_future):
         if loop is not None and loop is not futures._get_loop(coro_or_future):
-            raise ValueError('loop argument must agree with Future')
+            raise ValueError('The future belongs to a different loop than '
```

Comment:
What about "The future or coroutine"?

Reply:
Coroutines are handled by the `if` just above, no?

Reply:
hmmm I am not sure, but if I read the name `coro_or_future` I think that the ```_get_loop(coro_or_routine)``` handle both future and coroutine.

Reply:
hmmm I am not sure, but if I read the name coro_or_future I think that the _get_loop(coro_or_routine) handle both future and coroutine.

Reply:
Well, it seems that you're incorrect. :)